### PR TITLE
Set the `ServerName` field in the ClientHelloInfo when it was unset and a default SNI name is configured

### PR DESCRIPTION
**Pull request by ankon**
See discussion in slack: https://caddyserver.slack.com/archives/C4DMTEMGD/p1714689199952049

### CODE REVIEW

**Comment by ankon** [comment]
Related to: https://github.com/caddyserver/certmagic/tree/default-sni

File: modules/caddytls/connpolicy.go [hunk 1]
```diff
@@ -232,6 +232,9 @@ func (p *ConnectionPolicy) buildStandardTLSConfig(ctx caddy.Context) error {
 				cfg.CertSelection = p.CertSelection
 			}
 			cfg.DefaultServerName = p.DefaultSNI
+			if hello.ServerName == "" && p.DefaultSNI != "" {
```
Comment (ArturFortunato) [inline modules/caddytls/connpolicy.go:235]: The second validation is kinda redundant right? Worst case scenario you're just setting "" = ""

**Comment by mholt** [comment]
As discussed in Slack, this might be superseded by the default-sni branch in CertMagic which applies this a little more broadly at the cert-management layer.

**Comment by mholt** [comment]
If I understand correctly, the default-sni branch of CertMagic will supersede this, so I'll close this. (But correct me if I'm wrong and we can reopen it. :smiley: )

**Comment by ankon** [comment]
Mostly for linking things up: The "default-sni branch" is https://github.com/caddyserver/certmagic/pull/287, which is in certmagic and released with 0.21.1.

**Pull request closed without merging**